File: samples/snippets/csharp/VS_Snippets_WebNet/System.ComponentModel.DataAnnotations.MetadataTypeAttribute/CS/Customer.cs
//<Snippet1>
using System;
using System.Web.DynamicData;
using System.ComponentModel.DataAnnotations;

// <Snippet2>
[MetadataType(typeof(CustomerMetaData))]
public partial class Customer
{

   
}

// </Snippet2>

public class CustomerMetaData
{
    // <Snippet3>
    // Apply RequiredAttribute
    [Required(ErrorMessage = "Title is required.")]
    public object Title;
    // </Snippet3>

   
}

//</Snippet1>

File: samples/snippets/visualbasic/VS_Snippets_WebNet/System.ComponentModel.DataAnnotations.MetadataTypeAttribute/VB/Customer.vb
'<Snippet1>
Imports System
Imports System.Web.DynamicData
Imports System.ComponentModel.DataAnnotations

'<Snippet2>
<MetadataType(GetType(CustomerMetadata))> _
Partial Public Class Customer

End Class
'</Snippet2>

Public Class CustomerMetadata

    '<Snippet3>
    ' Apply RequitedAttribute.
    <Required(ErrorMessage:="Title is required.")> _
    Public Title As Object
    '</Snippet3>

   
End Class

'</Snippet1>